diesel/src/query_builder/ast_pass.rs

```diff
@@ -46,15 +56,41 @@ impl<'a, DB> AstPass<'a, DB> where
         }
     }
 
+    pub fn push_sql(&mut self, sql: &str) {
+        if let AstPassInternals::ToSql(ref mut builder) = self.internals {
+            builder.push_sql(sql);
+        }
+    }
+
+    pub fn push_identifier(&mut self, identifier: &str) -> QueryResult<()> {
+        if let AstPassInternals::ToSql(ref mut builder) = self.internals {
+            builder.push_identifier(identifier)?;
+        }
+        Ok(())
+    }
+
     pub fn push_bind_param<T, U>(&mut self, bind: &U) -> QueryResult<()> where
         DB: HasSqlType<T>,
         U: ToSql<T, DB>,
     {
-        if let AstPassInternals::CollectBinds(ref mut out) = self.internals {
-            out.push_bound_value(bind)?;
+        use self::AstPassInternals::*;
+        match self.internals {
+            ToSql(ref mut out) => out.push_bind_param(),
+            CollectBinds(ref mut out) => out.push_bound_value(bind)?,
+            _ => {}, // noop
```

Comment:
I tend to leave these to note that I am intentionally doing nothing in the else case. If we think they add little value, we should probably remove them across the entire code base